content/source/assets/javascripts/application.js

```diff
@@ -7,9 +7,51 @@
 
 //= require analytics
 
+// Set up terraform.io UI helpers
 document.addEventListener("turbolinks:load", function() {
     "use strict";
+
+    // In navigation sidebars, hide most sub-lists and reveal any that contain the
+    // current page. The a.current-page class is added during build by
+    // layouts/inner.erb.
     var docsSidebar = $("#docs-sidebar ul.nav.docs-sidenav");
     docsSidebar.find("ul.nav").addClass("nav-hidden");
     docsSidebar.find("li").has("a.current-page").addClass("active");
+
+
+    // On docs/content pages, add a hierarchical quick nav menu if there are any
+    // H2/H3/H4 headers.
+    // Check whether turbolinks cached the quick-nav elements:
+    if ($("#inner #inner-quicknav ul").length == 0) {
```

Comment:
Is this ever non-zero, in your experience? I haven't tried it locally, but I thought turbolinks cached pages _before_ calling `turbolinks:load` handlers, so I don't understand how it would ever save the generated quick-nav.

Reply:
Oh, I might have misunderstood the timing on the cache! Yeah, I've never managed to make it be non-zero. I guess I might as well not test for that after all.